### ISSUE TITLE
Add detailed and rich documentation

### ISSUE THREAD

**Issue by KiranArun** [KiranArun/Neural_Networks-demo#6]
Add equations, diagrams, animations, etc to all parts of the notebook.

Branch https://github.com/KiranArun/Neural_Networks-101-demo/tree/explanations 

Areas to cover:
- [x] Input data (images)
- [x] Output data (one-hot vectors)
- [x] Data set segmentation (training, validation, testing)
- [x] Batch size
- [x] Learning Rate
- [x] Epochs / Iterations
- [x] Input Placeholders (X, Y)
- [x] Trainable Variables (W, b)
- [x] Matrix Multiplication
- [x] Softmax
- [x] Cross entropy
- [x] Gradient descent / Back prop
- [x] Training process
